### PR TITLE
Tests writes and iteration with Loom

### PR DESCRIPTION
**Pull request by Tpt**

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -79,3 +79,5 @@ jobs:
         with:
           command: test
           args: --test loom --features=loom,instrumentation --release --verbose
+          env:
+            LOOM_MAX_PREEMPTIONS: 3
```
Comment (Tpt) [inline .github/workflows/rust.yml:83]: We limit here the number of context switch to keep the number of sequence of states that Loom checks to an amount compatible with a usable CI runtime.

File: src/table.rs [hunk 2]
```diff
@@ -155,6 +155,9 @@ impl Header {
 }
 
 pub struct Entry<B: AsRef<[u8]> + AsMut<[u8]>>(usize, B);
+#[cfg(feature = "loom")]
+pub type FullEntry = Entry<Vec<u8>>;
```
Comment (Tpt) [inline src/table.rs:159]: Bad hack because of the stack limitations introduced by Loom.

**Comment by arkpar** [comment]
Is it possible to increase the stack size with `ulimit` on the CI runner instead of changing the code?

**Comment by Tpt** [comment]
> Is it possible to increase the stack size with `ulimit` on the CI runner instead of changing the code?

I tried to play with it and with `RUST_MIN_STACK` that configures the spawned threads limit. Sadly I have not manage to get the tests to pass without either crashing or consuming all my laptop memory (32GB). It's quite unfortunate.

**Review by arkpar — APPROVED** [review]

File: tests/loom.rs [hunk 3]
```diff
@@ -6,31 +6,100 @@ use std::sync::Arc;
...
+			db.commit::<_, Vec<u8>>(vec![(0, vec![0], Some(vec![3])), (0, vec![1], Some(vec![3]))])
+				.unwrap();
+			db.enact_logs().unwrap();
 		});
 
 		t1.join().unwrap();
 		t2.join().unwrap();
+		t3.join().unwrap();
 
 		// Checks, that either T1 committed before T2 or reverse
```
Comment (cheme) [inline tests/loom.rs:56]: So IIUC here the assert_eq is not really what we want to test, comment may not be up to date. Would just put `Check state consistency`.

**Review by cheme — APPROVED** [review]

**Pull request merged**